dev/io.openliberty.microprofile.openapi.internal.common/src/io/openliberty/microprofile/openapi/internal/common/OpenAPIAppConfigProviderImpl.java

```diff
@@ -0,0 +1,189 @@
+                issuedBetaMessage = !issuedBetaMessage;
+            }
+        }
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    @Deprecated
+    public Optional<String> getIncludedModules() {
+
+        betaFenceCheck();
+        return includedString;
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    @Deprecated
+    public Optional<String> getExcludedModules() {
+        betaFenceCheck();
+        return excludedString;
+    }
+
+    @Override
+    public void registerAppConfigListener(OpenAPIAppConfigListener listener) {
+        openAPIAppConfigListeners.add(listener);
+    }
+
+    @Override
+    public void unregisterAppConfigListener(OpenAPIAppConfigListener listener) {
+        openAPIAppConfigListeners.add(listener);
+    }
+
+    /**
+     * Takes the server.xml values for ex/includedApps, ex/includedModules, and filters them down into
+     * the single comma seperated string that the existing codebase expects.
+     */
+    private void processModuleConfig(Map<String, Object> properties) {
+
+        //Read the server.xml
+        Optional<String[]> includedAppArray = Optional.ofNullable((String[]) properties.get(INCLUDE_APP_PROPERTY_NAME));
+        Optional<String[]> excludedAppArray = Optional.ofNullable((String[]) properties.get(EXCLUDE_APP_PROPERTY_NAME));
+        Optional<String[]> includedModulesArray = Optional.ofNullable((String[]) properties.get(INCLUDE_MODULE_PROPERTY_NAME));
+        Optional<String[]> excludedModulesArray = Optional.ofNullable((String[]) properties.get(EXCLUDE_MODULE_PROPERTY_NAME));
+
+        //Filter out apps from modules and modules from apps, condensing to a single comma separated string.
+        Optional<String> includedAppString = includedAppArray.map(this::appStringProcessing);
+        Optional<String> excludedAppString = excludedAppArray.map(this::appStringProcessing);
+        Optional<String> includedModulesString = includedModulesArray.map(this::moduleStringProcessing);
+        Optional<String> excludedModulesString = excludedModulesArray.map(this::moduleStringProcessing);
+
+        //The surprisingly complex code to combine two optionals, excluding null or blank strings
+        includedString = Stream.of(includedAppString,
+                                   includedModulesString).filter(Optional::isPresent).map(Optional::get).filter(StringUtils::isNotBlank).reduce((first, second) -> first + ','
+                                                                                                                                                                   + second);
+
+        excludedString = Stream.of(excludedAppString,
+                                   excludedModulesString).filter(Optional::isPresent).map(Optional::get).filter(StringUtils::isNotBlank).reduce((first, second) -> first + ','
+                                                                                                                                                                   + second);
```

Comment:
I dislike converting an array of Strings into a comma separated list here when we're going to convert it back into a list later.